sdk/core/azure_core_test/src/recording.rs

```diff
@@ -158,52 +177,68 @@ impl Recording {
-        let rng = self.rand.get_or_init(|| match self.test_mode {
-            TestMode::Live => ChaCha20Rng::from_entropy().into(),
-            TestMode::Playback => {
-                let variables = self
-                    .variables
-                    .read()
-                    .map_err(read_lock_error)
-                    .unwrap_or_else(|err| panic!("{err}"));
-                let seed: String = variables
-                    .get(NAME)
-                    .map(Into::into)
-                    .unwrap_or_else(|| panic!("random seed variable not set"));
-                let seed = base64::decode(seed)
-                    .unwrap_or_else(|err| panic!("failed to decode random seed: {err}"));
-                let seed = seed
-                    .first_chunk::<32>()
-                    .unwrap_or_else(|| panic!("insufficient random seed variable"));
-
-                ChaCha20Rng::from_seed(*seed).into()
-            }
-            TestMode::Record => {
-                let rng = ChaCha20Rng::from_entropy();
-                let seed = rng.get_seed();
-                let seed = base64::encode(seed);
+        let rng = self.rng();
+        let Ok(mut rng) = rng.lock() else {
+            panic!("failed to lock RNG");
+        };
 
-                let mut variables = self
-                    .variables
-                    .write()
-                    .map_err(write_lock_error)
-                    .unwrap_or_else(|err| panic!("{err}"));
-                variables.insert(NAME.to_string(), Value::from(Some(seed), None));
+        rng.gen()
+    }
 
-                rng.into()
+    /// Generate a random string with optional prefix.
+    ///
+    /// This will always be the OS cryptographically secure pseudo-random number generator (CSPRNG) when running live.
+    /// When recording, it will initialize from the OS CSPRNG but save the seed value to the recording file.
+    /// When playing back, the saved seed value is read from the recording to reproduce the same sequence of random data.
+    ///
+    /// # Examples
+    ///
+    /// Generate a random string.
+    ///
+    /// ```
+    /// # let recording = azure_core_test::Recording::with_seed();
+    /// let id = recording.random_string::<12>(Some("t")).to_ascii_lowercase();
+    /// # assert_eq!(id, "tpbwhbkhckmk");
+    /// ```
+    ///
+    /// # Panics
+    ///
+    /// Panics if the recording variables cannot be locked for reading or writing,
+    /// if the random seed cannot be encoded or decoded properly,
+    /// if `LEN` is 0,
+    /// or if the length of `prefix` is greater than or equal to `LEN`.
+    ///
+    /// ```should_panic
+    /// # let recording = azure_core_test::Recording::with_seed();
+    /// let vault_name = recording.random_string::<8>(Some("keyvault"));
+    /// ```
+    ///
+    pub fn random_string<const LEN: usize>(&self, prefix: Option<&str>) -> String {
+        struct NonZero<const N: usize>;
```

Comment:
I do this instead of a `NonZeroUsize` because you can't simply pass `0` as the argument. You'd have to do `NonZeroUsize::new(0)` and that's not a great DX.

Reply:
I was also trying to catch as many errors at compile time as I could. Alternatively, I could go back to the original signature I prototyped i.e., `random_string(&self, prefix: Option<&str>, len: usize) -> String` but then all the assertions are runtime assertions.